src/App.tsx

```diff
@@ -8,32 +9,40 @@ function getRandomName(): string {
 }
 
 export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+  const [clockName, setClockName] = useState('Clock-0');
+  const [hasClock, setHasClock] = useState(true);
+  const [timerId, setTimerId] = useState(0);
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  const contextmenuHandler = (event: MouseEvent) => {
```

Comment:
Usually these start with an action/verb. Since the only interest is in the right click and `contextmenu` pop-up is unwanted and prevented I would go for (imo):
```suggestion
  const handleRightClick = (event: MouseEvent) => {
```